sysstat-post-process

```diff
@@ -504,7 +504,52 @@ for my $log_file (sort readdir($dh)) {
             exit;
             }
         }
-    }
+    } elsif ($log_file =~ /^iostat-stdout.txt$/) {
+	my $log_fh = $log_file;
+        open(TMP,"$log_fh") || die "Could not open $log_fh\n";
+        my $separator = $/;
+        $/ = '';
+        my $ymd_timestamp_ms; # Epochtime in milliseconds
+
+        while (my($timestamp, $header, @devicelines) = split('\n', <TMP>)) {
+            foreach my $device_entry (@devicelines) {
+		my %iosample;
+                my $iodev = (split ' ', ${device_entry})[0];
+                (my $unused,
+                $iosample{$iodev}{'reads-sec'},
+                $iosample{$iodev}{'writes-sec'},
+                $iosample{$iodev}{'read-kb-sec'},
+                $iosample{$iodev}{'write-kb-sec'},
+                $iosample{$iodev}{'reads-merged-sec'},
+                $iosample{$iodev}{'writes-merged-sec'},
+                $iosample{$iodev}{'read-pct-merged'},
+                $iosample{$iodev}{'write-pct-merged'},
+                $iosample{$iodev}{'read-wait-avg-ms'},
+                $iosample{$iodev}{'write-wait-avg-ms'},
+                $iosample{$iodev}{'avg-queue-len'},
+                $iosample{$iodev}{'avg-read-req-size'},
+                $iosample{$iodev}{'avg-write-req-size'},
+		my $deprecated,
+                $iosample{$iodev}{'percent-utilization'}) = split(' ', $device_entry);
+
+		$ymd_timestamp_ms = `date +%s%N -d $timestamp -u` / 1000000;
```

Comment:
We should avoid forking as much as possible, putting this deep inside the loop means lots of overhead.

Reply:
I'll fix this.

Reply:
OK, I used a perl builtin for time manipulation.

```
                my $deprecated,
                $iosample{$iodev}{'percent-utilization'}) = split(' ', $device_entry);

                my $time_object = Time::Piece->strptime($timestamp,'%Y-%m-%dT%T%z');
                my %sample = ('end' => $time_object->epoch);

```

Reply:
> > There is also some weird indentation in multiple places in this patch.
> 
> I don't see this in my terminal, so maybe it is caused by using a tab? Not sure, but I will go look.

Yeah, its probably something odd like that.